lib/kubeclient/common.rb

```diff
@@ -286,6 +308,45 @@ def get_entities(entity_type, klass, resource_name, options = {})
       Kubeclient::Common::EntityList.new(entity_type, resource_version, collection)
     end
 
+    def get_entities_async(klass, resource_name, options = {})
```

Comment:
Is this method really async? Or it's more of a lazy execution? Usually `async` is a sign that the procedure is using threads, which I think is not the case here?

Reply:
it's async in that it's non-blocking. entities are `yield`ed to the block passed by the user as they are read from a streaming HTTP response